site/source/docs/api_reference/proxying.h.rst

```diff
@@ -0,0 +1,144 @@
+  proxied to a queue on a particular thread, that thread is notified to start
+  processing work from that queue if it is not already doing so.
+
+  Proxied work can only be completed on live thread runtimes, so users must
+  ensure either that all proxied work is completed before a thread exits or that
+  the thread exits with a live runtime, e.g. via
+  ``emscripten_exit_with_live_runtime`` to avoid dropped work.
+
+.. c:type:: em_proxying_ctx
+
+  An opaque handle to a currently-executing proxied task, used to signal the end
+  of the task.
+
+Functions
+---------
+
+.. c:function:: em_proxying_queue* em_proxying_queue_create()
+
+  Allocate a new proxying queue.
+
+.. c:function:: void em_proxying_queue_destroy(em_proxying_queue* q)
+
+  Free a proxying queue. The queue should not have any remaining queued work.
+
+.. c:function:: em_proxying_queue* emscripten_proxy_get_system_queue()
+
+  Get the queue used for proxying low-level runtime work. Work on this queue may
+  be processed at any time inside system functions, so it must be nonblocking
+  and safe to run at any time, similar to a native signal handler. User code
+  should generally not use this function.
+
+.. c:function:: void emscripten_proxy_execute_queue(em_proxying_queue* q)
+
+  Execute all the tasks enqueued for the current thread on the given queue. New
+  tasks that are enqueued concurrently with this execution will be executed as
+  well. This function returns once it observes an empty queue.
+
+.. c:function:: void emscripten_proxy_finish(em_proxying_ctx* ctx)
+
+  Signal the end of a task proxied with ``emscripten_proxy_sync_with_ctx``.
+
+.. c:function:: int emscripten_proxy_async(em_proxying_queue* q, pthread_t target_thread, void (\*func)(void*), void* arg)
+
+  Enqueue ``func`` to be called with argument ``arg`` on the given queue and
+  thread then return immediately. Returns 1 if the work was successfully
+  enqueued and the target thread notified or 0 otherwise.
+
+.. c:function:: int emscripten_proxy_sync(em_proxying_queue* q, pthread_t target_thread, void (\*func)(void*), void* arg)
+
+  Enqueue ``func`` to be called with argument ``arg`` on the given queue and
+  thread then wait for ``func`` to be executed before returning. Returns 1 if
+  the ``func`` was successfully completed and 0 otherwise.
+
+.. c:function:: int emscripten_proxy_sync_with_ctx(em_proxying_queue* q, pthread_t target_thread, void (\*func)(em_proxying_ctx*, void*), void* arg)
+
+  Enqueue ``func`` to be called with argument `arg`` on the given queue and
+  thread and wait for it to be executed and for the task to be marked finished
+  with ``emscripten_proxying_finish`` before returning. ``func`` need not call
+  ``emscripten_proxying_finish`` itself; it could instead store the context
+  pointer and call ``emscripten_proxying_finish`` at an arbitrary later time.
+  Returns 1 if the task was successfully completed and 0 otherwise.
```

Comment:
Just an idea, this could be described as "The same as `emscripten_proxy_sync`, except that [..diff..]".